core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java

```diff
@@ -86,74 +86,61 @@ public void setNumThreads(int numThreads) {
     this.numThreads = numThreads;
   }
 
+  @SuppressWarnings("unchecked")
   public GtfsFeedContainer loadAndValidate(
-      GtfsInput gtfsInput, ValidatorProvider validatorProvider, NoticeContainer noticeContainer) {
+      GtfsInput gtfsInput, ValidatorProvider validatorProvider, NoticeContainer noticeContainer)
+      throws InterruptedException {
     logger.atInfo().log("Loading in %d threads", numThreads);
     ExecutorService exec = Executors.newFixedThreadPool(numThreads);
 
     List<Callable<TableAndNoticeContainers>> loaderCallables = new ArrayList<>();
     Map<String, GtfsTableLoader<?>> remainingLoaders =
         (Map<String, GtfsTableLoader<?>>) tableLoaders.clone();
     for (String filename : gtfsInput.getFilenames()) {
-      GtfsTableLoader loader = remainingLoaders.remove(filename.toLowerCase());
+      GtfsTableLoader<?> loader = remainingLoaders.remove(filename.toLowerCase());
       if (loader == null) {
         noticeContainer.addValidationNotice(new UnknownFileNotice(filename));
       } else {
         loaderCallables.add(
             () -> {
-              InputStream inputStream = gtfsInput.getFile(filename);
               NoticeContainer loaderNotices = new NoticeContainer();
-              GtfsTableContainer tableContainer;
-              try {
-                tableContainer = loader.load(inputStream, validatorProvider, loaderNotices);
-              } catch (RuntimeException e) {
-                // This handler should prevent ExecutionException for
-                // this thread. We catch an exception here for storing
-                // the context since we know the filename here.
-                logger.atSevere().withCause(e).log("Runtime exception when loading %s", filename);
-                loaderNotices.addSystemError(
-                    new RuntimeExceptionInLoaderError(
-                        filename, e.getClass().getCanonicalName(), e.getMessage()));
-                // Since the file was not loaded successfully, we treat
-                // it as missing for continuing validation.
-                tableContainer = loader.loadMissingFile(validatorProvider, loaderNotices);
-              } finally {
-                inputStream.close();
```

Comment:
This line has been removed because wou wrote 
`try (InputStream inputStream = gtfsInput.getFile(filename)) {
...
}` earlier, right? Which will close our `InputStream`?

Reply:
Sure, that is called try-with-resources. https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html